content/authoring/kata/floating-point.md

```diff
@@ -0,0 +1,279 @@
+---
+kind: tutorial
+---
+
+# HOWTO Work with Floating-Point Numbers
+
+
+This article attempts to explain to kata authors what the most common problems and pitfalls encountered when creating kata are when working with floating-point numbers, and how to avoid them.
+Many freshly published kata turn out to have problems with handling precision of calculations, using incorrect assertions, or trying to work around the problems in an incorrect manner. This often causes many valid solutions to fail the tests. Errors related to floating-point values are often difficult to reproduce, so whenever an issue is reported, authors immediately close it as _"works for me, you are doing something wrong"_. Incorrectly handled floating-point numbers can also cause confusing errors and assertion messages.
+
+The article is not meant to be a comprehensive guide explaining what floating-point numbers are in general, how they are represented, how they work internally, etc. It's meant to be targeted mostly to authors of Codewars kata to help them create content of better quality and avoid traps and pitfalls leading to problems with precision loss, comparisons, etc.
+
+
+## What are floating-point numbers?
+
+Floating-point numbers come in many different shapes and colors, but one (or, if you prefer, two) of them is the most widespread, the most common, and, at the same time, the most problematic for newbies (and not only). Whenever you see or use anything called _"float"_, _"double"_, _"single-"_ or _"double-precision floating point number"_, _"IEEE-754 number"_, you need to be careful, because things can often turn out surprising and not as easy as one might think. Through the rest of the article, the term _"floating-point numbers"_ will refer specifically to **IEEE-754 double-precision values**, unless explicitly stated otherwise.
+
+Floating-point numbers are usually not specific to any particular language, system, or architecture. IEEE-754 numbers are widely adopted and used across many platforms, languages, and compilers. With some small exceptions, anything written in this article applies in the same way to Python, C, JavaScript, or Haskell.
+
+<details>
+
+<summary>
+"Why"
```

Comment:
Please don't do this. `<summary>` should be descriptive and shouldn't contain anything else like this note block.

Reply:
I used this only to collapse a section which I am not sure about. I am not sure if it is educational enough and should stay, or is it too detailed and should be removed.

If you guys think it should be removed, I will remove it. If you think it should stay, I will move it out of `<details>`. Or maybe you have some other idea?

Reply:
Unless you mean just the _"Why"_ "caption", it can be removed.

```suggestion
```

Reply:
No, this:
![image](https://user-images.githubusercontent.com/639336/108584156-f6e8bf80-72f3-11eb-897f-bbcd682c4add.png)

`<summary>` should be short and fit next to that triangle. Spec-wise it should only contain "phrasing content" (something that will work inside `<p>`). Browsers are lax because users don't write correct HTML :(

This huge summary is bad from usability point of view, try reproducing this scroll position

![image](https://user-images.githubusercontent.com/639336/108584252-a9208700-72f4-11eb-8a99-ee2e2275d11c.png)

then click the `<summary>`. The triangle points down, but it looks like nothing is showing up because it's happening below the block.

Reply:
I've removed the collapsible thing. Maybe now seeing how long the whole page gets it will be easier for reviewers to suggest whether the "why" part should stay or not.